returnn/tf/layers/basic.py

```diff
@@ -6655,8 +6662,12 @@ def get_value(self):
     :rtype: tf.Tensor
     """
     with tf.name_scope("loss_bin_ce"):
-      out = tf.nn.sigmoid_cross_entropy_with_logits(
-        logits=self.output_flat, labels=tf.cast(self.target_flat, self.output_flat.dtype))
+      target_flat = tf.cast(self.target_flat, self.output_flat.dtype)
+      if self._pos_weight is None:
+        out = tf.nn.sigmoid_cross_entropy_with_logits(logits=self.output_flat, labels=target_flat)
+      else:
```

Comment:
Do you need this branch? Wouldn't the other code just always work?

Reply:
The implementation is not exactly identical, so I think it is safer to keep it separated.

Reply:
Ah sorry, I didn't realized this is a different function.